### PR TITLE
Make navigation options all public!

### PR DESCRIPTION
**Pull request by cdroulers**
Add at least one test that uses the options!

### CODE REVIEW

**Comment by cdroulers** [comment]
Slightly irrelevant, but couldn't `testCert.cer` be commited along the code since it's a self-signed test certificate? It would make contributing slightly easier!

**Comment by kblok** [comment]
I could try to upload my certificate, but as it's a self-signed I don't know if it would work. That's why I created a tutorial instead https://github.com/kblok/puppeteer-sharp/blob/master/CONTRIBUTING.md#getting-setup :p

**Comment by kblok** [comment]
Could you create an issue, asking for the certificate?

File: lib/PuppeteerSharp/NavigationOptions.cs [hunk 1]
```diff
@@ -2,7 +2,8 @@
 {
     public class NavigationOptions
     {
-        public int? Timeout { get; internal set; }
-        public string[] WaitUntil { get; internal set; }
+        public int? Timeout { get; set; }
+
```
Comment (kblok) [inline lib/PuppeteerSharp/NavigationOptions.cs]: Let's remove the extra line

File: lib/PuppeteerSharp.Tests/Puppeteer/PuppeteerLaunchTests.cs [hunk 2]
```diff
@@ -38,6 +38,19 @@ public class PuppeteerLaunchTests : PuppeteerBaseTest
...
 
+        [Fact]
+        public async Task ShouldWorkInRealLifeWithOptions()
+        {
+            var options = TestConstants.DefaultBrowserOptions();
+
+            using (var browser = await PuppeteerSharp.Puppeteer.LaunchAsync(options, TestConstants.ChromiumRevision))
+            using (var page = await browser.NewPageAsync())
+            {
+                var response = await page.GoToAsync("https://www.google.com", new NavigationOptions() { Timeout = 5000, WaitUntil = new[] { "networkidle0" } });
```
Comment (kblok) [inline lib/PuppeteerSharp.Tests/Puppeteer/PuppeteerLaunchTests.cs]: Can we make this multi-line?

```cs 
var response = await page.GoToAsync("https://www.google.com", new NavigationOptions
{ 
    Timeout = 5000, 
    WaitUntil = new[] { "networkidle0" }
});
```

Comment (cdroulers) [reply lib/PuppeteerSharp.Tests/Puppeteer/PuppeteerLaunchTests.cs]: Stylecop recommends putting each parameter on its own line in that case, is that ok?

```csharp
                var response = await page.GoToAsync(
                    "https://www.google.com",
                    new NavigationOptions()
                    {
                        Timeout = 5000,
                        WaitUntil = new[] { "networkidle0" }
                    });
```

File: lib/PuppeteerSharp.Tests/Puppeteer/PuppeteerLaunchTests.cs [hunk 3]
```diff
@@ -38,6 +38,19 @@ public class PuppeteerLaunchTests : PuppeteerBaseTest
...
+                Assert.Equal(response.Status.ToString(), "OK");
```
Comment (Meir017) [inline lib/PuppeteerSharp.Tests/Puppeteer/PuppeteerLaunchTests.cs]: use `HttpStatusCode.OK` instead

Comment (cdroulers) [reply lib/PuppeteerSharp.Tests/Puppeteer/PuppeteerLaunchTests.cs]: Will change all in this file!

**Comment by kblok** [comment]
I'd like your approval @Meir017

**Comment by kblok** [comment]
Congratulations @cdroulers!

**Comment by cdroulers** [comment]
@kblok As for the certificate, I figured out how to create one anyway and it worked, I just realized that it probably wouldn't out of the box because the certificate needs to be trusted, otherwise we'd get HTTPS errors. So instructions are fine, really!

**Pull request merged**